RecoLocalCalo/HcalRecAlgos/src/parseHFPhase1AlgoDescription.cc

```diff
@@ -30,6 +30,12 @@ parseHFPhase1AlgoDescription(const edm::ParameterSet& ps)
         const bool rejectAllFailures =
             ps.getParameter<bool>("rejectAllFailures");
 
+        float minChargeForUndershoot = -10000.f, minChargeForOvershoot = -10000.f;
+        if (ps.existsAs<double>("minChargeForUndershoot"))
```

Comment:
Well, suggest a solution then. Note that multiple reconstruction codes in CMSSW split their functionality between modules and algorithms. Discarding C++ inheritance of algorithms in favor of having "fillDescriptions" is a truly brain dead idea.